apps/view/src/logger.ts

```diff
@@ -0,0 +1,52 @@
+import {useState, useEffect, useMemo} from 'react'
+
+import {LogLevel, Logger, LogHandler} from './types'
+
+const STORAGE_KEY = 'logLevel'
+const DEFAULT_LEVEL = process.env.NODE_ENV !== 'production' ? 'debug' : 'warn'
+const LEVELS: Array<LogLevel> = ['debug', 'info', 'warn', 'error']
+
+const createLogHandler = (logLvl: LogLevel, lvl: LogLevel): LogHandler =>
+  LEVELS.indexOf(lvl) >= LEVELS.indexOf(logLvl)
+    ? (msg, ...meta) => console[lvl](`${lvl}: ${msg}`, ...meta)
+    : () => {}
+
+export function useLogger(): Logger {
```

Comment:
I'm not sure this hook is behaving as desired; I can't enable debug logs in production